merlin/models/tf/inputs/embedding.py

```diff
@@ -150,6 +150,182 @@ def from_config(cls, config):
+            Default is None 
+    Examples
+    --------
+    .. code-block:: python
+
+    Notes
+    -----
+    """
+    def __init__(self,
+        dim: int,
+        *col_schemas: ColumnSchema,
+        vocab_sizes: list,
+        initializer: Union[str, tf.Tensor, list] = "uniform",
+        use_dynamic_variable = False,
+        localized = None,
+        name = None,
+        **kwargs
+        ):
+        super(SOKEmbedding, self).__init__(
+             dim,
+             *col_schemas,
+             trainable=trainable,
+             name=name,
+             dtype=dtype,
+             **kwargs)
+        self._embedding_vec_size = dim
+        self._vocab_sizes = vocab_sizes
+        self._use_dynamic_variable = use_dynamic_variable
+        self._localized = localized
+        if self._localized is None and self._use_dynamic_variable == False :
+            prefix_sum = []
+            offset = 0
+            for i in range(len(vocab_sizes)):
+                prefix_sum.append(offset)
+                offset += self._vocab_sizes[i]
+            prefix_sum = np.array(prefix_sum, dtype=np.int64).reshape(1, -1)
+            self._vocab_prefix_sum = tf.constant(prefix_sum)
+            print("[Info] Total vocabulary size:", offset)
+
+            if isinstance(initializer, str):
+                self._var = sok.Variable(
+                    shape=[offset, self._embedding_vec_size], initializer=tf.keras.initializers.get(initializer), dtype=tf.float32
+                )
+            else:
+                self._var = sok.Variable(initializer)
+        else:
+            self._vars = []
+            for i in range(len(vocab_sizes)):
+                if _use_dynamic_variable:
+                    if isinstance(initializer, str): 
+                        v = sok.DynamicVariable(dimension=self._embedding_vec_size, initializer=initializer)
+                    else:
+                        indices = tf.convert_to_tensor(initializer[i][0])
+                        values = tf.convert_to_tensor(initializer[i][1])
+                        sok.assign(v, indices, values)
+                elif self._localized is not None:
+                    if isinstance(initializer, str): 
+                        v = sok.Variable(
+                            shape=[self._vocab_sizes[i], self._embedding_vec_size],
+                            initializer=tf.keras.initializers.get(intializer),
+                            dtype=tf.float32,
+                            mode="localized:%d" % self._localized[i],
+                        )
+                    else:
+                        v = sok.Variable(
+                            initializer[i],
+                            mode="localized:%d" % self._localized[i],
+                        )
+                else:
+                    raise ValueError("Wrong Configuration!!!")
+                self._trainable_weights.append(v)
+                self._vars.append(v)
+
+    def call(self, inputs, training=True):
```

Comment:
Should we throw in exception when a dense-tensor is provided here?

Reply:
sok.lookup_sparse can accept Sparse and dense-tensor, but both of them should have a combiner
292 def lookup_sparse(params, sp_ids, combiners):
293     """
294     Abbreviated as ``sok.experiment.lookup_sparse``.
295
296     Peform fused sparse lookup on the given embedding ``params``. This function
297     is similar to the ``tf.nn.embedding_lookup_sparse``, but with two differences:
298
299         - It can do distributed lookup.
300         - It can accept multiple params and multiple sp_ids to do fused lookup at once,
301           which brings performance benifits.
302
303     Parameters
304     ----------
305     params: list, tuple
306             a list or tuple of trainable *sok.Variable*.
307     sp_ids: list, tuple
308             a list or tuple of tf.SparseTensor or tf.RaggedTensor.
309     combiners: list, tuple
310             a list or tuple of string to specify the combiner of each lookup.